src/services/SearchResults.cpp

```diff
@@ -103,36 +103,40 @@ static const char* ComparisonString(ComparisonType nCompareType)
     }
 }
 
-static unsigned int GetValue(const unsigned char* pBuffer, unsigned int nOffset, ComparisonVariableSize nSize)
+_Success_(return)
+_NODISCARD _CONSTANT_FN GetValue(_In_z_ const unsigned char* const pBuffer,
+                                 _In_   unsigned int nOffset,
+                                 _In_   MemSize nSize) noexcept
```

Comment:
You lost the `static unsigned int`, which makes it unclear that this method is only used by this file, and what the return type is. I see the `_CONSTANT_FN` expands to `constexpr auto`. `constexpr` is supposed to be used to indicate it's compile-time constant, but this method is called with three variables, so nothing about this function is compile-time constant. Besides confusing me, what is the purpose of this change?

Also note that `pBuffer` is not a null terminated string so the `_In_z_` attribute may be inappropriate. It's a pointer to a block of memory.

Reply:
It'll go back to being just static, and the annotation for pBuffer will change to just _In_ since it's not a null terminated string.